x/nft/approvals.go

```diff
@@ -58,22 +58,34 @@ func (a ApprovalOptions) Equals(o ApprovalOptions) bool {
 	return a.Immutable == o.Immutable && a.Count == o.Count && a.UntilBlockHeight == o.UntilBlockHeight
 }
 
-//TODO: decide what we need to validate here and what before
-//This requires all the model-specific actions to be passed here
-func (m Approvals) Validate(actions ...string) error {
-	actions = append(actions, []string{ActionUpdateDetails, ActionTransfer}...)
-	actionMap := make(map[string]struct{}, 0)
-	for _, action := range actions {
-		actionMap[action] = struct{}{}
+func (a ApprovalOptions) Validate() error {
+	if a.Count == 0 || a.Count < UnlimitedCount {
+		return errors.ErrInternal("Approval count should either be unlimited or above zero")
 	}
+	return nil
+}
 
+//This requires all the model-specific actions to be passed here
+//TODO: Not sure I'm a fan of array of maps, but it makes sense
+//given we validate using protobuf enum value maps
+func (m Approvals) Validate(actionMaps ...map[string]int32) error {
 	for action := range m {
-		if _, ok := actionMap[action]; !ok {
+
+		withinImpl := func() bool {
```

Comment:
nice. i like the actionMaps so far.
the int32 is just a placeholder or it has a meaning (enum)?
i guess i just gotta keep reading...

Reply:
yup, i saw this later... going through code reviews in default alphabetical order is not always optimal.